### PR TITLE
Implemented SQLite log storage for Objective-C; added tests.

### PR DESCRIPTION
**Pull request by abohomol**

### CODE REVIEW

File: client/client-multi/client-objective-c/Kaa/log/SQLiteLogStorage.m [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+@property (nonatomic) int32_t currentRecordCount;
+
+/**
+ * Format: <int32_t, int64_t> as key-value wrapped with NSNumber;
+ * key:     bucket id;
+ * value:   bucket size.
+ */
+@property (nonatomic, strong) NSMutableDictionary *consumedMemoryMap;
+
+- (void)openDB:(NSString *)dbPath;
```
Comment (sergiyk0) [inline client/client-multi/client-objective-c/Kaa/log/SQLiteLogStorage.m]: Better rename as openDBAtPath:

Comment (abohomol) [inline client/client-multi/client-objective-c/Kaa/log/SQLiteLogStorage.m]: [Fixed](https://github.com/kaaproject/kaa/pull/557/commits/8ca753c9e639d5259607697e610797f765594776).

File: client/client-multi/client-objective-c/Kaa/log/PersistentLogStorageConstants.h [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef PersistentLogStorageConstants_h
+#define PersistentLogStorageConstants_h
+
+#define DEFAULT_DB_NAME             @"kaa_logs.db"
```
Comment (sergiyk0) [inline client/client-multi/client-objective-c/Kaa/log/PersistentLogStorageConstants.h]: Shouldn't it have extension .sqlite?

Comment (abohomol) [inline client/client-multi/client-objective-c/Kaa/log/PersistentLogStorageConstants.h]: For real, it doesn't matter or does it?

**Comment by rasendubi** [comment]
Please, reopen against develop

**Pull request closed without merging**